lib/iruby/command.rb

```diff
@@ -46,69 +80,27 @@ def check_version
     def run_ipython
       check_version
 
-      dir = @args.grep(/\A--iruby-dir=.*\Z/)
-      @args -= dir
-      dir = dir.last.to_s.sub(/\A--profile=/, '')
-      dir = ENV['IRUBYDIR'] || IRUBYDIR if dir.empty?
-      dir = File.expand_path(dir)
-      ENV['IPYTHONDIR'] = dir
-
-      if @args.size == 3 && @args[0] == 'profile' && @args[1] == 'create'
-        profile = @args[2]
-      else
-        profile = @args.grep(/\A--profile=.*\Z/).last.to_s.sub(/\A--profile=/, '')
-        profile = 'default' if profile.empty?
-      end
-
-      create_profile(dir, profile)
-
       # We must use the console to launch the whole 0MQ-client-server stack
-      @args = %w(console --no-banner) + @args if @args.first.to_s !~ /\A\w+\Z/
+      @args = %w(console --no-banner) + @args if @args.first.to_s !~ /\A\w/
+      register_iruby_kernel if %w(console qtconsole notebook).include? @args.first
+      @args += %w(--kernel ruby) if %w(console qtconsole).include? @args.first
 
       Kernel.exec('ipython', *@args)
     end
 
-    def create_profile(dir, profile)
-      profile_dir = File.join(dir, "profile_#{profile}")
-      unless File.directory?(profile_dir)
-        puts "Creating profile directory #{profile_dir}"
-        `ipython profile create #{Shellwords.escape profile}`
-      end
-
-      kernel_cmd = []
-      kernel_cmd << ENV['BUNDLE_BIN_PATH'] << 'exec' if ENV['BUNDLE_BIN_PATH']
-
-      if RUBY_PLATFORM =~ WINDOWS_REGEXP
-        kernel_cmd += [RbConfig.ruby, File.expand_path('../../../bin/iruby', __FILE__)].map{|path| path.gsub('/', '\\\\\\') }
-      else
-        kernel_cmd << File.expand_path($0)
-      end
-      kernel_cmd << 'kernel' << '{connection_file}'
-
-      kernel_cmd = "c.KernelManager.kernel_cmd = #{kernel_cmd.inspect}"
-      Dir[File.join(profile_dir, '*_config.py')].each do |path|
-        content = File.read(path)
-        content << kernel_cmd unless content.gsub!(/^c\.KernelManager\.kernel_cmd.*$/, kernel_cmd)
-        File.write(path, content)
-      end
-
-      begin
-        static_dir = File.join(profile_dir, 'static')
-        target_dir = File.join(File.dirname(__FILE__), 'static')
-        begin
-          old_target = File.readlink(static_dir) rescue nil
-        rescue NotImplementedError
-          FileUtils.cp_r(target_dir, static_dir)
-          return
-        end
+    def register_iruby_kernel
+      ipython_register_file.parent.mkpath
+      ipython_register_file.write <<-EOF.chomp
+{
+"argv": [ "#{File.expand_path $0}", "kernel", "{connection_file}" ],
+"display_name": "Ruby",
+"language": "ruby"
+}
+EOF
+    end
```

Comment:
It would be better if we check here if the file exists and not overwrite it in this case. Maybe the user modified it?